Build a macOS bundle for Intel and ARM

mrled opened the pull request:
Build a macOS bundle on Intel and ARM. This generates a nice double-clickable `Lan Mouse.app` application.

I've tested the bundling locally and it does work for me, but the Github Actions might require some adjustment.

Let me know if anything needs adjustment, I am not very familiar with the rust/cargo world so maybe I made some mistakes :)

Includes an icns file created from the existing icon. I created it like this:

```sh
cd lan-mouse-gtk/resources
mkdir icon.iconset
magick convert -density 256x256 -background none de.feschber.LanMouse.svg icon.iconset/icon_512x512@2x.png
magick convert icon.iconset/icon_512x512@2x.png -resize 16x16 icon.iconset/icon_16x16.png
magick convert icon.iconset/icon_512x512@2x.png -resize 32x32 icon.iconset/icon_16x16@2x.png
magick convert icon.iconset/icon_512x512@2x.png -resize 32x32 icon.iconset/icon_32x32.png
magick convert icon.iconset/icon_512x512@2x.png -resize 64x64 icon.iconset/icon_32x32@2x.png
magick convert icon.iconset/icon_512x512@2x.png -resize 128x128 icon.iconset/icon_128x128.png
magick convert icon.iconset/icon_512x512@2x.png -resize 256x256 icon.iconset/icon_128x128@2x.png
magick convert icon.iconset/icon_512x512@2x.png -resize 256x256 icon.iconset/icon_256x256.png
magick convert icon.iconset/icon_512x512@2x.png -resize 512x512 icon.iconset/icon_256x256@2x.png
magick convert icon.iconset/icon_512x512@2x.png -resize 512x512 icon.iconset/icon_512x512.png 
iconutil -c icns icon.iconset
```

mrled commented:
That makes sense. Should I make it buildable locally too, outside of GitHub actions?

I could add a shell script like `scripts/makeicns.sh` or something like that. Maybe that script could be integrated into the Rust build, although I wouldn't want to break the Rust build on non-macOS platforms that don't have an `iconutil` command.

feschber commented:
Yeah that's fine for now. I don't think it makes sense to add it to the build.rs because that is meant to generate runtime dependencies and the icon pack is not one.
So invoking `scripts/makeicns.sh` and then `cargo bundle` is fine.

mrled commented:
Ok done, might want to squash this on merge so that the icns binary file I added in the first commit and removed in the second doesn't stay in git history forever.

feschber commented:
Alright, lets see if it works ;)

feschber commented:
Oh btw, we need to add that stuff to the release actions as well so that it gets released automatically.

I can do that myself if you want or you take a look at the pre-release / tagged-release.yml files.

feschber commented:
Now available at https://github.com/feschber/lan-mouse/releases!
Since I do not own a mac myself: Could you confirm it's working correctly?

And again thank you very much for your contribution!

mrled commented:
It works for me 🎊

However, by default macOS quarantines downloaded files, and prevents you from opening them:
 
<img width="372" alt="Screenshot 2025-02-21 at 11 15 54" src="https://github.com/user-attachments/assets/8fe2812d-1ee1-40c0-960b-0c96b76bb1a0" />

You can get around this by unquarantining the application after unzipping it:

```sh
xattr -rd com.apple.quarantine "/Applications/Lan Mouse.app"
```

I think that codesigning is supposed to help with this, either allowing you to right click -> open or to double click as normal and accept a warning that the app is built by an unknown developer, but I don't know off the top of my head exactly what the steps for that are.

feschber commented:
yeah I will need to look into signing packages for Windows as well. And same goes for requesting accessibility permissions.
